src/Products/Models/Products.php

```diff
@@ -106,4 +112,254 @@ public function beforeCreate()
             $this->short_slug = $this->slug;
         }
     }
+
+    /**
+     * Add a product to a specify category.
+     *
+     * @param ModelsCategories $category
+     *
+     * @return ProductCategory
+     */
+    public function addCategory(ModelsCategories $category) : ProductCategory
+    {
+        return ProductCategory::findFirstOrCreate([
+            'conditions' => 'products_id = :products_id: AND categories_id = :categories_id:',
+            'bind' => [
+                'products_id' => $this->getId(),
+                'categories_id' => $category->getId(),
+            ]
+        ], [
+            'categories_id' => $category->getId(),
+            'products_id' => $this->getId(),
+        ]);
+    }
```

Comment:
@kaioken Maybe move those methods to Products/Products an call it from here so that way is gonna look bonito.

public addCategory($category)
{
 return Products::addCategory($this, $category);
}

Reply:
will do sr, thanks